## APOE apolipoprotein E [ Homo sapiens (human) ]



Introduction

What is APOE ?

Why is APOE one of the strongest genetics risk factors for Alzheimer's disease ?

How do APOE variants influence Alzheimer's disease risk?

Manual Database Investigation [NCBI, Ensembl, ClinVar]

NCBI Summary2 image

[APOE - NCBI](https://www.ncbi.nlm.nih.gov/gene/348)

## Phase 2
### Biological and Database Exploration of APOE

In [45]:
apoe_human_info = {
    "Gene Symbol" : "APOE",
    "Official Full Name" : "apolipoprotein E",
    "NCBI Gene ID" : "348",
    "Gene type" : "protein coding",
    "Ensembl ID" : "ENSG00000130203",
    "HGNC ID" : "HGNC:613",
    "Other Names" : "AD2, LPG, APO-E, ApoE4, LDLCQ5",
    "Organism" : "Homo sapiens",
    "Protein Isoforms" : "5",
    "Transcripts Counts" : "5",
    "Chromosome": "19",
    "Location": "19q13.32",
    "Exon count": "6",
    "Intron count" : "5"  #Exon - 1
}

In [13]:
import pandas as pd

df = pd.DataFrame(list(apoe_human_info.items()), columns= ["Identifier Type", "Value"])

df.style

,Identifier Type,Value
0,Gene Symbol,APOE
1,Official Full Name,apolipoprotein E
2,NCBI Gene ID,348
3,Gene type,protein coding
4,Ensembl ID,ENSG00000130203
5,HGNC ID,HGNC:613
6,Other Names,"AD2, LPG, APO-E, ApoE4, LDLCQ5"
7,Organism,Homo sapiens
8,Protein Isoforms,5
9,Transcripts Counts,5


---


Genomic regions, transcripts, and products2.png

---

### 1.4 Reproducing Manual Findings Programmically

 ### Gene ID 

In [46]:
from Bio import Entrez

Entrez.email = "sahelsepehri99@gmail.com"

handle = Entrez.esearch(
    db= "gene",
    term = "APOE[Gene] AND Homo sapiens [Organism]"
)

record = Entrez.read(handle)
gene_id = record ["IdList"][0]
print("Gene ID :", gene_id)

Gene ID : 348


---

Gene Summary

In [47]:
handle = Entrez.esummary(
    db="gene",
    id=gene_id,
    retmode="xml"
)

summary = Entrez.read(handle)

doc = summary["DocumentSummarySet"]["DocumentSummary"][0]


data = {
    "Gene Name": doc["Name"],
    "Description": doc["Description"],
    "Chromosome": doc["Chromosome"],
    "Map Location": doc["MapLocation"],
    "Aliases": doc["OtherAliases"]
}

df = pd.DataFrame([data])
df



,Gene Name,Description,Chromosome,Map Location,Aliases
0,APOE,apolipoprotein E,19,19q13.32,"AD2, APO-E, ApoE4, LDLCQ5, LPG"


---

Sequence Retrieval

mRNA Sequence

This Gene has 5 mRNA but mRNA NM_000041.4 is a main mRNA of this gene.

>NM_000041.4 APOE [organism=Homo sapiens] [GeneID=348] [transcript=2]
CTACTCAGCCCCAGCGGAGGTGAAGGACGTCCTTCCCCAGGAGCCGACTGGCCAATCACAGGCAGGAAGA
TGAAGGTTCTGTGGGCTGCGTTGCTGGTCACATTCCTGGCAGGATGCCAGGCCAAGGTGGAGCAAGCGGT
GGAGACAGAGCCGGAGCCCGAGCTGCGCCAGCAGACCGAGTGGCAGAGCGGCCAGCGCTGGGAACTGGCA
CTGGGTCGCTTTTGGGATTACCTGCGCTGGGTGCAGACACTGTCTGAGCAGGTGCAGGAGGAGCTGCTCA
GCTCCCAGGTCACCCAGGAACTGAGGGCGCTGATGGACGAGACCATGAAGGAGTTGAAGGCCTACAAATC
GGAACTGGAGGAACAACTGACCCCGGTGGCGGAGGAGACGCGGGCACGGCTGTCCAAGGAGCTGCAGGCG
GCGCAGGCCCGGCTGGGCGCGGACATGGAGGACGTGTGCGGCCGCCTGGTGCAGTACCGCGGCGAGGTGC
AGGCCATGCTCGGCCAGAGCACCGAGGAGCTGCGGGTGCGCCTCGCCTCCCACCTGCGCAAGCTGCGTAA
GCGGCTCCTCCGCGATGCCGATGACCTGCAGAAGCGCCTGGCAGTGTACCAGGCCGGGGCCCGCGAGGGC
GCCGAGCGCGGCCTCAGCGCCATCCGCGAGCGCCTGGGGCCCCTGGTGGAACAGGGCCGCGTGCGGGCCG
CCACTGTGGGCTCCCTGGCCGGCCAGCCGCTACAGGAGCGGGCCCAGGCCTGGGGCGAGCGGCTGCGCGC
GCGGATGGAGGAGATGGGCAGCCGGACCCGCGACCGCCTGGACGAGGTGAAGGAGCAGGTGGCGGAGGTG
CGCGCCAAGCTGGAGGAGCAGGCCCAGCAGATACGCCTGCAGGCCGAGGCCTTCCAGGCCCGCCTCAAGA
GCTGGTTCGAGCCCCTGGTGGAAGACATGCAGCGCCAGTGGGCCGGGCTGGTGGAGAAGGTGCAGGCTGC
CGTGGGCACCAGCGCCGCCCCTGTGCCCAGCGACAATCACTGAACGCCGAAGCCTGCAGCCATGCGACCC
CACGCCACCCCGTGCCTCCTGCCTCCGCGCAGCCTGCAGCGGGAGACCCTGTCCCCGCCCCAGCCGTCCT
CCTGGGGTGGACCCTAGTTTAATAAAGATTCACCAAGTTTCACGCA

In [49]:
from Bio import SeqIO

handle = Entrez.efetch(
    db="nucleotide",
    id="NM_000041.4",
    rettype="fasta",
    retmode="text"
)

mrna_record = SeqIO.read(handle, "fasta")

print(mrna_record.description)

print("Sequence length:", len(mrna_record.seq))

print(mrna_record.seq[:100])


SeqIO.write(mrna_record, "APOE_mRNA.fasta", "fasta")



NM_000041.4 Homo sapiens apolipoprotein E (APOE), transcript variant 2, mRNA
Sequence length: 1166
CTACTCAGCCCCAGCGGAGGTGAAGGACGTCCTTCCCCAGGAGCCGACTGGCCAATCACAGGCAGGAAGATGAAGGTTCTGTGGGCTGCGTTGCTGGTCA


1

In [50]:
print("C :", mrna_record.count("C"))
print("G :", mrna_record.count("G"))
print("A :", mrna_record.count("A"))
print("T :", mrna_record.count("T"))

C : 373
G : 432
A : 211
T : 150


---

### GC Content

https://en.wikipedia.org/wiki/GC-content

In [32]:
from Bio.SeqUtils import gc_fraction

gc = gc_fraction(mrna_record.seq) * 100
print("GC Content:", gc)


GC Content: 69.03945111492281


---

proteins Sequence

This Gene has 5 proteins but >NP_000032.1 is a main protein of this gene.

>NP_000032.1 APOE [organism=Homo sapiens] [GeneID=348] [isoform=b precursor]
MKVLWAALLVTFLAGCQAKVEQAVETEPEPELRQQTEWQSGQRWELALGRFWDYLRWVQTLSEQVQEELL
SSQVTQELRALMDETMKELKAYKSELEEQLTPVAEETRARLSKELQAAQARLGADMEDVCGRLVQYRGEV
QAMLGQSTEELRVRLASHLRKLRKRLLRDADDLQKRLAVYQAGAREGAERGLSAIRERLGPLVEQGRVRA
ATVGSLAGQPLQERAQAWGERLRARMEEMGSRTRDRLDEVKEQVAEVRAKLEEQAQQIRLQAEAFQARLK
SWFEPLVEDMQRQWAGLVEKVQAAVGTSAAPVPSDNH


In [34]:

handle = Entrez.efetch(
    db="protein",
    id="NP_000032.1",
    rettype="fasta",
    retmode="text"
)

protein_record = SeqIO.read(handle, "fasta")

print(protein_record.description)
print("Sequence length:", len(protein_record.seq))
print(protein_record.seq[:100])

SeqIO.write(protein_record, "APOE_protein.fasta", "fasta")


NP_000032.1 apolipoprotein E isoform b precursor [Homo sapiens]
Sequence length: 317
MKVLWAALLVTFLAGCQAKVEQAVETEPEPELRQQTEWQSGQRWELALGRFWDYLRWVQTLSEQVQEELLSSQVTQELRALMDETMKELKAYKSELEEQL


1

---

### Protein Information

- RNA title:	mRNA-apolipoprotein E, transcript variant 2
- Protein title:	apolipoprotein E isoform b precursor
- Protein comment:	isoform b precursor is encoded by transcript variant 2
- Merged features: NM_000041.4 and NP_000032.1
- Location:	44,905,796..44,909,393	
- Span on NC_000019.10: 3,598 nt
- Aligned length:	1,166 nt
- CDS length:	954 nt
- Protein length:	317 aa


- mRNA RefSeq : NM_000041.4
- Protein Refseq : NP_000032.1


Gene Symbol: APOE
Full name: apolipoprotein E



Transcript variants
Protein isoforms
Gene length

---

Variant Name
Clinical Significance
Associated disease
Isoform
Effect



PubMed
APOE Alzheimer's review

Title
Year
Journal
D0I

---

References 

- [NCBI](https://www.ncbi.nlm.nih.gov/)
- [Pandas Documentation](https://pandas.pydata.org/docs/index.html)
- [Accessing NCBI’s Entrez databases - biopython](https://biopython.org/docs/latest/Tutorial/chapter_entrez.html)
- [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)
- [BLAST](https://blast.ncbi.nlm.nih.gov/)